ablySpec/RealtimeClientChannel.swift

```diff
@@ -609,6 +609,65 @@ class RealtimeClientChannel: QuickSpec {
 
             }
 
+            // RTL8
+            context("unsubscribe") {
+
+                // RTL8a
+                it("with no arguments unsubscribes the provided listener to all messages if subscribed") {
+                    let client = ARTRealtime(options: AblyTests.commonAppSetup())
+                    defer { client.close() }
+
+                    let channel = client.channels.get("test")
+
+                    var listener: ARTEventListener?
+                    waitUntil(timeout: testTimeout) { done in
+                        listener = channel.subscribe { message in
+                            expect(message.data as? String).to(equal("message"))
+                            done()
+                        }
+                        channel.publish(nil, data: "message")
+                    }
+
+                    channel.unsubscribe(listener!)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        channel.publish(nil, data: "message") { errorInfo in
+                            expect(errorInfo).to(beNil())
+                            // If `unsubscribe` fails then the test suite will raise "Done closure's was called multiple times."
```

Comment:
This test is unnecessarily complex because of this.  Why are we testing publish on line 628 at all. Surely we just add a listener, and the unsubscribe the listener.  We have other tests to test that the listener registration works, so if we don't test something that this test is not about, you can instead raise an error in the listener block because we never expect it to be called.